src/services/storage.cpp

```diff
@@ -170,6 +172,85 @@ void Storage::handleTrivialError(const leveldb::Status &status) {
   }
 }
 
+void Storage::saveBlock(Block &block) {
+  // Meta
+  BytesBuilder meta_header_builder;
+
+  string compression_algo_type_str;
+  if (block.compression_algo_type == CompressionAlgorithmType::LZ4)
+    compression_algo_type_str = "LZ4";
+  else
+    compression_algo_type_str = "NONE";
+
+  meta_header_builder.append(compression_algo_type_str);
+  meta_header_builder.append(block.header_length);
```

Comment:
```suggestion
  meta_header_builder.append(block.header_length,4);
```
int 계열의 경우 길이를 지정해주면, 뒤에서부터 딱 그만큼만 들어갑니다. 여기에서는 4바이트를 쓰니 4라고 지정해주어야 합니다. 그렇지 않으면 8바이트가 들어갑니다.

Reply:
어쨌든, header_length는 JSON을 만들어 stringify해서 압축하기 전에는 알기 어렵습니다. 이 부분이 227번 줄 뒤에 있어야할 것 같습니다.

Reply:
그리고 meta_header는 바이너리 데이터 구조인데,

1) 압축 알고리즘 (1byte)
2) 3)까지의 길이 (4byte)
3) block_header (json 타입)을 stringify하고 이를 압축한 데이터
4) 1)+2)+3)에 대한 서명

을 포함해야 합니다.